src/commands/new.ts

```diff
@@ -67,58 +67,44 @@ export default class New extends Command {
     flags: Flags<typeof New.flags>;
     args: { providerName?: string; prompt?: string };
   }): Promise<void> {
+    const ux = UX.create();
     const { providerName, prompt } = args;
 
-    let providerJson: ProviderJson;
-    let profile: { source: string; scope?: string; name: string };
-    const tasks = new Listr<{
-      providerName: string | undefined;
-      prompt: string | undefined;
-    }>([
-      {
-        title: 'Checking input arguments',
-        task: async ctx => {
-          checkPrompt(ctx.prompt, { userError });
-        },
-      },
-      {
-        title: 'Loading provider definition',
-        task: async ctx => {
-          providerJson = await resolveProviderJson(ctx.providerName, {
-            userError,
-          });
-        },
-      },
-      {
-        title: 'Preparing Comlink interface your use case',
-        enabled: () => providerJson !== undefined,
-        task: async ctx => {
-          // TODO: should take also user error?
-          profile = await newProfile(
-            {
-              providerJson,
-              // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-              prompt: ctx.prompt!,
-              // TODO: add quiet flag to ctx?
-              options: { quiet: flags.quiet },
-            },
-            { logger }
-          );
-        },
-      },
+    ux.start('Checking input arguments');
+
+    checkPrompt(prompt, { userError });
+
+    ux.succeed('Input arguments checked');
+
+    ux.start('Loading provider definition');
+    const providerJson = await resolveProviderJson(providerName, {
+      userError,
+    });
+
+    ux.succeed('Provider definition loaded');
+
+    ux.start('Preparing Comlink interface your use case');
```

Comment:
issue: The message is hard to understand. I would omit the new `Comlink` term, created profile has `.profile` externsion, also the input iin map command is called profile. My suggestion:

```suggestion
    ux.start('Creating profile for your use case');
```